### PR TITLE
Fix: CalendarDatePicker does not handle validation correctly

### LINKED ISSUE #7307: Validation is not working properly in `CalendarDatePicker`
**Describe the bug**
If you validate the `SelectedDate`-property in you ViewModel the error message doesn't get shown

**To Reproduce**

1. Go to https://github.com/timunie/Tims.Avalonia.Samples/tree/main/src/CalendarDatePickerValidationIssue
2. Run the sample
3. See the valiadtion works for `TextBox` but not for `CalendarDatePicker`

**Expected behavior**
validation works the same as for `TextBox`

**Screenshots**
image

**Desktop (please complete the following information):**

- OS: [Windows]
- Version [0.10.11]

**Additional context**
I have a solution and will add the PR for this soon.

### PR DESCRIPTION
**Pull request by timunie**
## What does the pull request do?
Fix validation for `SelectedDate`-property

## What is the current behavior?
See #7307 

## What is the updated/expected behavior with this PR?
Validation works the same way as in `TextBox`

## How was the solution implemented (if it's not obvious)?
- changed how the validation works in the control
- changes the `Styles` to render the validation properly
- Updated the demo

## Checklist

- [ ] Added unit tests (if possible)? ► I think this is not needed here
- [X] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation ► I think this is not needed here

## Breaking changes
As the validation was there before I think this is not a breaking change

## Obsoletions / Deprecations
None

## Fixed issues
Fixes #7307

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0017721-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Themes.Default/CalendarDatePicker.xaml [hunk 1]
```diff
@@ -96,7 +98,7 @@
                    DataValidationErrors.Errors="{TemplateBinding (DataValidationErrors.Errors)}"
                    VerticalContentAlignment="{TemplateBinding VerticalContentAlignment}"
                    HorizontalContentAlignment="{TemplateBinding HorizontalContentAlignment}"
-                   Grid.Column="0"/>
+                   Grid.Column="0" />
 
           <Button Name="PART_Button"
```
Comment (danwalmsley) [inline src/Avalonia.Themes.Default/CalendarDatePicker.xaml:103]: Please revert all the whitespace changes in this file, it will mess up the history and we wont be able to trace who made what changes later. I cant see which is the actual change here if any...

<DataValidationErrors> element must be present here somewhere.

File: src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml [hunk 2]
```diff
@@ -19,125 +19,131 @@
...
+    <Setter Property="Background" Value="{DynamicResource CalendarDatePickerBackground}" />
+    <Setter Property="Foreground" Value="{DynamicResource CalendarDatePickerForeground}" />
+    <Setter Property="BorderBrush" Value="{DynamicResource CalendarDatePickerBorderBrush}" />
+    <Setter Property="BorderThickness" Value="{DynamicResource CalendarDatePickerBorderThemeThickness}" />
     <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
     <Setter Property="HorizontalAlignment" Value="Left" />
     <Setter Property="VerticalAlignment" Value="Center" />
-    <Setter Property="Padding" Value="4"/>
-    
+    <Setter Property="Padding" Value="4" />
```
Comment (danwalmsley) [inline src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml:32]: Please revert unrelated whitespace changes here also.

**Review by danwalmsley — CHANGES_REQUESTED** [review]

File: src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml (hunk 2 above)
Comment (timunie) [reply src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml]: I force pushed a fixed commit for this.

File: src/Avalonia.Themes.Default/CalendarDatePicker.xaml (hunk 1 above)
Comment (timunie) [reply src/Avalonia.Themes.Default/CalendarDatePicker.xaml]: > DataValidationErrors element must be present here somewhere. 

I use the one ot the TextBox as this looks really neat imo (sorry for the german message): 
image

The whitespaces are removed now.

**Comment by timunie** [comment]
Let me show some Preview (As the message is auto-generated it shows in German language, sorry for that): 

**Default Theme:**
image


**Fluent Theme:**
image

Happy coding
Tim

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0017736-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Review by danwalmsley — APPROVED** [review]

**Comment by danwalmsley** [comment]
Great job!
